spec/features/user_index_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.feature 'User index page', type: :feature do
+  before do
+    @user1 = User.create(name: 'tony',
+                         photo: 'https://image.com/image.jpg', bio: 'Text for Bio', posts_counter: 0)
+
+    3.times do |i|
+      Post.create(title: "Post #{i + 1}", author: @user1, comments_counter: 0, likes_counter: 0)
+    end
+
+    @user2 = User.create(name: 'Tumaini',
+                         photo: 'https://www.istockphoto.com/photo.jpg', bio: 'Text for Bio', posts_counter: 0)
+  end
+
+  scenario 'I can see the username of all other users' do
+    visit users_path
+
+    expect(page).to have_content('Marco')
+    expect(page).to have_content('Tumaini')
+  end
+
+  scenario 'I can see the profile picture for each user' do
+    visit users_path
+
+    expect(page).to have_css("img[src*='#{@user1.photo}']")
+    expect(page).to have_css("img[src*='#{@user2.photo}']")
+  end
+
+  scenario 'I can see the number of posts each user has written' do
+    visit users_path
+
+    expect(page).to have_content('Number of posts: 3')
+    expect(page).to have_content('Number of posts: 0')
+  end
+
+  scenario "When I click on a user, I am redirected to that user's show page" do
+    visit users_path
+    find('.user-row-link', text: 'Marco').click
+
+    expect(page.current_path).to eq(user_path(@user1))
+  end
```

Comment:
- [ ] You have some failing scenarios:

![image](https://github.com/ANTOINE1128/Blog-app/assets/77069011/377394ea-9ce0-4067-b006-4d1dbf452bef)

Please consider fixing them! 🤓 🔧